tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -75,6 +75,24 @@ def test_debugger_attribute():
     assert dev._debugger is None
 
 
+def test_snapshot_multiprocessing():
+    """DefaultQubit2 cannot execute tapes with Snapshot if `max_workers` is not `None`"""
+    dev = DefaultQubit2(max_workers=2)
+
+    tape = qml.tape.QuantumScript(
+        [
+            qml.Snapshot(),
+            qml.Hadamard(wires=0),
+            qml.Snapshot("very_important_state"),
+            qml.CNOT(wires=[0, 1]),
+            qml.Snapshot(),
+        ],
+        [qml.expval(qml.PauliX(0))],
+    )
+    with pytest.raises((RuntimeError, qml.DeviceError)):
+        dev.execute(tape)
```

Comment:
Maybe we should actually have an integration test here?

dev = qml.devices.experimental.DefaultQubit2(max_workers=2)

@qml.qnode(dev)
def circuit():
    qml.Snapshot("tag")
    qml.Hadamard(wires=0)
    qml.CNOT(wires=[0, 1])
    qml.Snapshot()
    return qml.expval(qml.PauliX(0) + qml.PauliY(0))

with pytest.raises(qml.DeviceError, match="Debugging with ``Snapshots`` is not available with multiprocessing."):
    qml.snapshots(circuit)()

Reply:
Yes, I'll add this and keep the other one for raw quantum scripts.